src/randoop/operation/Operation.java

```diff
@@ -4,62 +4,166 @@
 import java.util.List;
 
 import randoop.ExecutionOutcome;
+import randoop.reflection.ReflectionPredicate;
 import randoop.sequence.Variable;
 
 /**
- * Represents a template for a statement that can be part of a sequence, for example, a particular
- * method call, constructor call, or primitive value declaration.
+ * Operation represents the constructs that can occur in a statement as part of 
+ * a test sequence.
+ * These include method calls, constructor calls, field accesses, enum constant 
+ * values, or primitive values. 
+ * They are used both as symbols in constructing statements as part of tests, 
+ * and as a computational action in reflective execution of test sequences.
  * <p>
- * 
- * An implementing class C should additionally provide the following:
- * 
+ * The concept of an operation comes from logic and universal algebra, where 
+ * operations are used to build terms, which represent values. 
+ * (To dive into the formality, see, e.g., the 
+ * <a href="https://en.wikipedia.org/wiki/Term_algebra">Term Algebra</a> Wikipedia entry.)
+ * <p>
+ * An operation op has a type-signature 
+ * op: [T1, T2, ..., Tn] &rarr; T, where [T1, T2, ..., Tn] is the list of input 
+ * types, and T is the output type. 
+ * The input types are represented by an ordered list of 
+ * {@link Class} objects, and the output type is a single {@link Class} object. 
```

Comment:
It's true that the output type is a single Class object, but it's also the case that the operation may side-effect some of the inputs and thus may be viewed as having additional possible outputs.  Is that fact expressed?